substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1094,86 +1094,21 @@ pub mod pallet {
 			#[pallet::compact] value: BalanceOf<T>,
 		) -> DispatchResultWithPostInfo {
 			let controller = ensure_signed(origin)?;
-			let unlocking =
```

Comment:
github doesn't allow me to add comment on the non changed line `1090` so adding here.

Add chill weight to the call so in worst case it always adds chill weight. 
```
		#[pallet::call_index(2)]
		#[pallet::weight(
            T::WeightInfo::withdraw_unbonded_kill(SPECULATIVE_NUM_SPANS).saturating_add(T::WeightInfo::unbond())
			.saturating_add(T::WeightInfo::Chill()))
        ]
```
Additional info: Before any transaction this fund is reserved, and any difference in the PostInfo is refunded. So this should always contain the worst case weight.